test/DynamoCoreTests/Engine/LinkedListOfListTest.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections;
+using System.Reflection;
+using Dynamo.Engine;
+using NUnit.Framework;
+
+namespace Dynamo.Tests.Engine
+{
+    [TestFixture]
+    class LinkedListOfListTest : DynamoModelTestBase
+    {
+        /// <summary>
+        /// This test method will execute the next methods from the LinkedListOfList class:
+        /// IEnumerator IEnumerable.GetEnumerator()
+        /// IEnumerator<List<T>> IEnumerable<List<T>>.GetEnumerator()
+        /// public List<TKey> GetKeys()
+        /// </summary>
+        [Test]
+        [Category("UnitTests")]
+        public void LinkedListOfListEnumeratorsTest()
+        {
+            //Arrange
+            LinkedListOfList<Guid, string> nodes = new LinkedListOfList<Guid, string>();
+            nodes.AddItem(Guid.NewGuid(), "test1");
+            nodes.AddItem(Guid.NewGuid(), "test2");
+            nodes.AddItem(Guid.NewGuid(), "test3");
+
+            //Act
+            var nodesKeys = nodes.GetKeys();
+
+            //We need to use reflection since the GetEnumerator methods are not used (no references) and are private to the LinkedListOfList class
+            MethodInfo dynMethod = nodes.GetType().GetMethod("System.Collections.IEnumerable.GetEnumerator", BindingFlags.NonPublic | BindingFlags.Instance);
+            var enumerator1 = (IEnumerator)dynMethod.Invoke(nodes, null);
```

Comment:
Why are we testing private methods? It would be good to test the public/internal methods that are making use of the private ones.

Reply:
@aparajit-pratap 
When checking sonarqube, coverage section for the file Engine/LinkedListOfList.cs you will see that both methods (GetEnumerator) doesn't have coverage. I agree with you about using public methods that calls the private methods (in most cases i use them) but if you check in Visual Studio you will see that the references to that function won't call this method, for example this is one reference:

    IEnumerator IEnumerable.GetEnumerator()
    {
                return GetEnumerator();//This doesn't call the one in LinkedListOfList.cs the the coverage will be the same
     }

Then i didn't find any method that call the LinkedListOfList.GetEnumerator() and because is private using reflection was the only way that i found to do it (another one could be creating a child class and call the method).

Thanks

Reply:
I think that another option could be to remove that code, if it's private and not referenced, I think is unnecessary code that we can get rid off.

Probably another task could be to clean those classes when we find a case like that @QilongTang

Reply:
I don't think that we can get rid of these methods either as they are required by the interface. In that case, I'm okay with keeping this test.